### PR TITLE
Committee types

### PR DESCRIPTION
**Pull request by denalimarsh**
Refactor Committee to an interface that supports `MemberCommittee` and `TokenCommittee`

### CODE REVIEW

**Comment by karzak** [comment]
The interface approach sounds good. 

Only note is that `VoteThreshold` still needs to be used by both committee types and can be included in the base committee type. This is because a coin voting committee needs both 1) the quorum needed for the tallying to even take place and 2) the `VoteThreshold`, or the percentage of coins that must vote YES for a proposal to pass.

File: x/committee/types/committee.go [hunk 1]
```diff
@@ -13,52 +13,148 @@ import (
...
+		c.ID, c.Description, c.Permissions,
+		c.VoteThreshold.String(), c.ProposalDuration.String(),
+		c.TallyOption,
+	)
+}
+
+//
+type MemberCommittee struct {
+	BaseCommittee `json:"base_committee" yaml:"base_committee"`
+	Members       []sdk.AccAddress `json:"members" yaml:"members"`
```
Comment (karzak) [inline x/committee/types/committee.go:122]: `Members` should be added to the base type, as this defines who is allowed to make proposals. Given that, I think `MemberCommittee` can just be an alias for the base committee type.

**Review by karzak — CHANGES_REQUESTED** [review]

File: x/committee/types/committee.go [hunk 2]
```diff
@@ -73,32 +121,121 @@ func (c Committee) Validate() error {
...
+		Quorum:     quorum,
+		TallyDenom: tallyDenom,
+	}
+}
+
+// GetType returns the type of the committee
+func (c TokenCommittee) GetType() string { return TokenCommitteeType }
+
+// Validate validates the committee's fields
+func (c TokenCommittee) Validate() error {
```
Comment (karzak) [inline x/committee/types/committee.go:209]: This method should return `c.BaseCommittee.Validate()`

File: x/committee/types/committee.go [hunk 3]
```diff
@@ -73,32 +121,121 @@ func (c Committee) Validate() error {
...
+			Permissions:      permissions,
+			VoteThreshold:    threshold,
+			ProposalDuration: duration,
+			TallyOption:      tallyOption,
+		},
+	}
+}
+
+// GetType returns the type of the committee
+func (c MemberCommittee) GetType() string { return MemberCommitteeType }
```
Comment (karzak) [inline x/committee/types/committee.go:179]: Define `Validate` method and have it return `c.BaseCommittee.Validate()`

File: x/committee/types/committee.go [hunk 4]
```diff
@@ -13,32 +13,60 @@ import (
 
 const MaxCommitteeDescriptionLength int = 512
 
-// ------------------------------------------
-//				Committees
-// ------------------------------------------
+type TallyOption int
+
+const (
```
Comment (karzak) [inline x/committee/types/committee.go:18]: Add comment to each of the tally options that describes them.

FPTH - Votes are tallied each block and the proposal passes as soon as the vote threshold is reached
Deadline - Votes are tallied exactly once, when the deadline time is reached

File: x/committee/types/committee.go [hunk 5]
```diff
@@ -13,32 +13,60 @@ import (
...
+	FirstPastThePost TallyOption = iota // 0
+	Deadline         TallyOption = iota // 1
+)
+
+const (
```
Comment (karzak) [inline x/committee/types/committee.go:23]: Add descriptive comments for each committee type

**Review by karzak — CHANGES_REQUESTED** [review]
Left a few nits, should be mergeable after changes

File: x/committee/types/committee.go (hunk 5 above)
Comment (denalimarsh) [reply x/committee/types/committee.go:23]: https://github.com/Kava-Labs/kava/pull/899/commits/5ff4eeb6591dcf2040bb900af67fdfe6b2165c98

File: x/committee/types/committee.go (hunk 4 above)
Comment (denalimarsh) [reply x/committee/types/committee.go:18]: https://github.com/Kava-Labs/kava/pull/899/commits/5ff4eeb6591dcf2040bb900af67fdfe6b2165c98

File: x/committee/types/committee.go (hunk 3 above)
Comment (denalimarsh) [reply x/committee/types/committee.go:179]: https://github.com/Kava-Labs/kava/pull/899/commits/5ff4eeb6591dcf2040bb900af67fdfe6b2165c98

File: x/committee/types/committee.go (hunk 2 above)
Comment (denalimarsh) [reply x/committee/types/committee.go:214]: https://github.com/Kava-Labs/kava/pull/899/commits/5ff4eeb6591dcf2040bb900af67fdfe6b2165c98

**Review by karzak — APPROVED** [review]
LGTM

**Pull request merged**